spring-amqp/src/main/java/org/springframework/amqp/core/Address.java

```diff
@@ -39,14 +39,25 @@
  */
 public class Address {
 
+	/**
+	 * Use this value in {@code RabbitTemplate#setReplyAddress(String)} to use a temporary reply queue
+	 * for each request, regardless of whether the broker supports direct reply-to.
+	 */
+	public static final String TEMPORARY_REPLY_QUEUE_TOKEN = "##TEMPORARY_QUEUE##";
```

Comment:
We could, but my aim to was to make this have minimal impact - this solution requires no schema changes (to `<rabbit:template/>`); it also allows us to backport it to 1.5.x. (Our general rule is to not make XML schema changes in maintenance releases).

If we make it a boolean, it would only be available using Java configuration - or you'd have to set it programmatically after the context is initialized (with 1.5). We'd do the schema/parser change in 1.6.

To be honest, I doubt many people will use this (which is why I changed the logic to use direct reply-to when it's available in the first place). I don't really understand the use case for creating a temporary queue for each request when a much more efficient mechanism is available.

Can you explain your use case further? If you can convince me that this should be a first-class option, I'm happy to reconsider.